BranchMetrics/android-branch-deep-linking-attribution
# fix:SDK-2025 -- [Android] Remove unused code

## Reference
SDK-2025 -- [Android] Remove unused code

## Description
Removing more code that has fallen into disuse.
- ServerRequestRegisterClose
- ~~FB app link~~ may still be used, researching.
- Referral/Content URL builders
- Unused NetworkCallback interface
- Deprecated request paths
    - Credits
- Defunct CI scripts
    - GitLab
- A notice for a bug that has since been fixed
    - https://bugs.chromium.org/p/chromium/issues/detail?id=484815
    - https://bugs.chromium.org/p/chromium/issues/detail?id=638672

## Testing Instructions
Run through a test app, especially opening and closing as we're removing the post. No behavior should change.

## Risk Assessment [`LOW`]



- [✅] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style

cc @BranchMetrics/saas-sdk-devs for visibility.
